src/main/java/org/traccar/protocol/Tlt2hProtocolDecoder.java

```diff
@@ -148,7 +150,11 @@ protected Object decode(
 
         String status = parser.next();
 
-        String[] messages = sentence.substring(sentence.indexOf('\n') + 1).split("\r\n");
+        String[] messages = Arrays.stream(sentence.substring(sentence.indexOf('\n') + 1).split("\r\n"))
```

Comment:
You're still doing the same thing. I think you should do something like this:
```
sentence.substring(sentence.indexOf('\n') + 1, sentence.endsWith("##") ? sentence.length() - 4 : sentence.length())
```

Reply:
That's much more elegant! Thanks for suggesting.